### ISSUE TITLE
Support for fuzzy searches?

### ISSUE THREAD

**Issue by toastdriven** [django-haystack/django-haystack#164]
Solr supports fuzzy searching ("roam~" - http://lucene.apache.org/java/2_4_0/queryparsersyntax.html) which a lot of users have been looking for (perhaps better than wildcard search).

Investigate if Xapian/Whoosh support this and implement if so.

**Comment by webyneter** [comment]
Am I right that today there's no way to make Solr-driven fuzzy search work via haystack/python?

**Comment by acdha** [comment]
@webyneter No - it works in every backend which supports it. What we don't have is an abstraction layer which tries to cover the differences between the different search engines, since that's a fairly hard problem for all but the most basic use-cases.

http://django-haystack.readthedocs.io/en/v2.6.0/inputtypes.html#raw

**Comment by webyneter** [comment]
@acdha, yep, I've just stumbled upon Field Lookups section in the docs. Thanks for the quick response.

**Comment by SalahAdDin** [comment]
Maybe will be good make a way of extends haystack for people that need add more features, it makes easy implements this kind of features by self.

**Comment by acdha** [comment]
@SalahAdDin That's exactly what the input types system is designed to do: see http://django-haystack.readthedocs.io/en/v2.6.0/inputtypes.html

**Comment by divonelnc** [comment]
I am using haystack with Whoosh as a backend. Is there any way to perform a fuzzy search with those? `SearchQuerySet().filter(content=search_string)` does not seem to handle fuzziness.

**Comment by prafgup** [comment]
@divonelnc Any update regarding implementing fuzziness with Whoosh?
